.github/workflows/ci-build.yml

```diff
@@ -145,9 +145,15 @@ jobs:
           cname: ${{ env.DOCUMENTATION_CNAME }}
           token: ${{ secrets.GITHUB_TOKEN }}
 
+
+  doc-index-dev:
+    name: "Deploy dev docs index"
+    #if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    needs: docs-build
```

Comment:
This should run after deploying the docs @Revathyvenugopal162 - please put a ``needs`` stage

Reply:
dev docs is deploying along with docs build and its already there in line 153., if you prefer i can break the dev docs deploy steps too.

Reply:
I see. Can't we keep it all together in the same stage?

Reply:
Split deployment and indexing into separate stages. Have them run after doc build

Reply:
Ok, i will split it.